AutomatedTesting/Gem/PythonTests/atom_renderer/atom_hydra_scripts/hydra_AtomEditorComponents_GlobalSkylightIBLAdded.py

```diff
@@ -0,0 +1,159 @@
+    import azlmbr.editor as editor
+    import azlmbr.legacy.general as general
+    import azlmbr.math as math
+
+    from editor_python_test_tools.editor_entity_utils import EditorEntity
+    from editor_python_test_tools.utils import Report, Tracer, TestHelper as helper
+
+    with Tracer() as error_tracer:
+        # Test setup begins.
+        # Setup: Wait for Editor idle loop before executing Python hydra scripts then open "Base" level.
+        helper.init_idle()
+        helper.open_level("Physics", "Base")
+
+        # Test steps begin.
+        # 1. Create a Global Skylight (IBL) entity with no components.
+        global_skylight_name = "Global Skylight (IBL)"
+        global_skylight_entity = EditorEntity.create_editor_entity_at(
+            math.Vector3(512.0, 512.0, 34.0), global_skylight_name)
+        Report.critical_result(Tests.global_skylight_creation, global_skylight_entity.exists())
+
+        # 2. Add Global Skylight (IBL) component to Global Skylight (IBL) entity.
+        global_skylight_component = global_skylight_entity.add_component(global_skylight_name)
+        Report.critical_result(
+            Tests.global_skylight_component, global_skylight_entity.has_component(global_skylight_name))
+
+        # 3. UNDO the entity creation and component addition.
+        # Requires 3 UNDO calls to remove the Entity completely.
+        for x in range(4):
+            general.undo()
+        general.idle_wait_frames(1)
+        Report.result(Tests.creation_undo, not global_skylight_entity.exists())
+
+        # 4. REDO the entity creation and component addition.
+        # Requires 3 REDO calls to match the previous 3 UNDO calls.
+        for x in range(4):
+            general.redo()
+        general.idle_wait_frames(1)
+        Report.result(Tests.creation_redo, global_skylight_entity.exists())
+
+        # 5. Enter/Exit game mode.
+        helper.enter_game_mode(Tests.enter_game_mode)
+        general.idle_wait_frames(1)
+        helper.exit_game_mode(Tests.exit_game_mode)
+
+        # 6. Test IsHidden.
+        global_skylight_entity.set_visibility_state(False)
+        is_hidden = editor.EditorEntityInfoRequestBus(bus.Event, 'IsHidden', global_skylight_entity.id)
+        Report.result(Tests.is_hidden, is_hidden is True)
+
+        # 7. Test IsVisible.
+        global_skylight_entity.set_visibility_state(True)
+        is_visible = editor.EditorEntityInfoRequestBus(bus.Event, 'IsVisible', global_skylight_entity.id)
+        Report.result(Tests.is_visible, is_visible is True)
+
+        # 8. Set the Diffuse Image asset on the Global Skylight (IBL) entity.
+        global_skylight_diffuse_image_property = "Controller|Configuration|Diffuse Image"
+        diffuse_image_path = os.path.join("LightingPresets", "greenwich_park_02_4k_iblskyboxcm.exr.streamingimage")
+        diffuse_image_asset = asset.AssetCatalogRequestBus(
+            bus.Broadcast, "GetAssetIdByPath", diffuse_image_path, math.Uuid(), False)
```

Comment:
repeated: you can use the Asset class utility. Check for other instances in the rest of the code